src/setup/rgw.py

```diff
@@ -14,77 +14,220 @@
+
+        try:
+            hostname = cortx_gconf.get(f'node>{Rgw._machine_id}>hostname')
+            host_ip = socket.gethostbyname(hostname)
+            ceph_tmpl_idx = 'ceph_conf_tmpl'
+            # Load template ceph.conf file and updated required field.
+            kv_list = [
+                ('global>fsid', str(uuid.uuid1())),
+                ('global>mon host', host_ip)]
+            Rgw._update_config(ceph_tmpl_idx, ceph_conf_tmpl_url, kv_list)
+
+            # Copy ceph_conf_tmpl_file to /etc/ceph/ dir.
+            if not os.path.exists(ceph_conf_path):
+                os.mkdir(ceph_conf_path)
+            shutil.copyfile(ceph_conf_tmpl_file, ceph_conf_file)
+            Log.info(f'{ceph_conf_tmpl_file} config copied to {ceph_conf_file}.')
+
+            # create admin keyring.
+            cmd = f"sudo ceph-authtool --create-keyring {ceph_admin_keyring_file} \
+                --gen-key -n client.admin --cap mon 'allow *' --cap osd 'allow *'"
+            _, err, rc, = SimpleProcess(cmd).run()
+            if rc != 0:
+                raise RgwSetupError(errno.EINVAL, '"{cmd}" failed with error {err}.')
+        except Exception as e:
+            raise RgwSetupError(errno.EINVAL, f'Error ocurred while fetching node ip, {e}')
+
+        Log.info('Prepare phase completed.')
 
         return 0
 
     @staticmethod
-    def config(config_path: str):
-        """
-        Performs configurations.
-        """
+    def config(cortx_gconf: MappedConf):
+        """Performs configurations."""
+
+        Log.info('Config phase started.')
 
+        # TODO: Inside [client.rgw.<hostname>] section and
+        # add motr,hax endpoints and rgw client related parameters.
+        Log.info('Config phase completed.')
         return 0
 
     @staticmethod
-    def init(config_path: str):
-        """
-        Perform initialization.
-        """
-
+    def init(cortx_gconf: MappedConf):
+        """Perform initialization."""
+
+        Log.info('Init phase started.')
+        # TODO: Create admin user.
+        # admin user should be created only on one node.
+        # 1. While creating admin user, global lock created in consul kv store.
+        # (rgw_consul_index, cortx>rgw>volatile>rgw_lock, machine_id)
+        # 2. Before creating admin user.
+        #    a. Check for rgw_lock in consul kv store.
+        #    b. Create user only if lock value is None/machine-id.
+
+        rgw_lock = False
+        rgw_lock_key = 'component>rgw>volatile>rgw_lock'
+        rgw_consul_idx = 'rgw_consul_idx'
+        # Get consul url from cortx config.
+        consul_url = Rgw._get_consul_url(cortx_gconf)
+        # Check for rgw_lock in consul kv store.
+        Log.info('Checking for rgw lock in consul kv store.')
+        Conf.load(rgw_consul_idx, consul_url)
+
+        while(True):
+            try:
+                rgw_lock_val = Conf.get(rgw_consul_idx, rgw_lock_key)
+                Log.info(f'rgw_lock value - {rgw_lock_val}')
```

Comment:
Please check if this should be outside while loop.

Reply:
Please check if below conditional statements should be outside try-except.

Reply:
we are reading and updating the rgw_lock_key value in each iteration.
Exception may occur while reading/updating value in consul, to handle that I have added this code in try-except block.